extensions/ql-vscode/src/remote-queries/remote-queries-interface.ts

```diff
@@ -129,6 +135,31 @@ export class RemoteQueriesInterfaceManager {
     });
   }
 
+  private async openFile(filePath: string) {
+    try {
+      const textDocument = await workspace.openTextDocument(filePath);
+      await Window.showTextDocument(textDocument, ViewColumn.One);
+    } catch (error) {
+      void showAndLogWarningMessage(`Could not open file: ${filePath}`);
+    }
+  }
+
+  private async openVirtualFile(text: string) {
+    try {
+      const params = new URLSearchParams({
+        queryText: encodeURIComponent(SHOW_QUERY_TEXT_MSG + text)
+      });
+      const uri = Uri.parse(
+        `remote-query:query-text.ql?${params.toString()}`,
+        true
+      );
+      const doc = await workspace.openTextDocument(uri);
+      await Window.showTextDocument(doc, { preview: false });
```

Comment:
Is this read-only? It's possible to make happen, but I forget how.

Reply:
Indeed, it's read-only! That's what the `remote-query` prefix and the corresponding [provideTextDocumentContent](https://github.com/github/vscode-codeql/pull/1041/files#diff-8ab69a78d0894d504f6993ec074207e3afe8cbcdddf051ba50b98dd2cecc980cR996) function do 😃